go-controller/pkg/clustermanager/secondary_network_cluster_manager.go

```diff
@@ -12,6 +14,11 @@ import (
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
 )
 
+const (
+	// Maximum secondary network IDs that can be generated. An arbitrary value is chosen.
+	maxSecondaryNetworkIDs = 4096
```

Comment:
@jcaamano can you shed some light into how exactly does this limit the number of secondary networks ? 1:1 assoc between these IDs and the network? 

Hm iirc, we had a use case for secondary ovn-k networks to scale past the number of vlans for an interface. 

This may impact us; is there a change of having a larger number of IDs ? (since it's arbitrary ...)

Reply:
I guess a geneve TLV with the network name instead of VNI is out of the question here :P

Reply:
> @jcaamano can you shed some light into how exactly does this limit the number of secondary networks ? 1:1 assoc between these IDs and the network?

Yes, so this would mean a max of 4096 secondary networks.

Reply:
It could be increased I guess but to which value. What is the max value we would be happy with?

Reply:
Up to a max of 65532 since the VNI is 24bit and the base VNI for transist switches is 16711683, maybe

Reply:
Can anything bad / does it have any cost if we set the max then ?

Reply:
Nothing bad would happen.  There is some cost though in the bitmap management .
I think it would be O(n) for allocating the next bit.


Shall I set this value to 65532 ?  or for now just restrict to 1000 ?

Reply:
I don't know. Does 65532 represent all available space? Does that space also need to account for the possible node ID values?

It is strange because the LS tunnel key is documented as range 1 to 16,777,215. And the LSP tunnel key is documented as range 1 to 32,767. I don't see how all that fits a 24 bit VNI.

I think having an idea of a maximum number of networks would help @maiqueb. If it needs to be as big as 65532 then so be it. But if we can cap that at anything lower it would be somewhat helpful.

Reply:
That's because in the geneve options we store both the inport tunnely key and outport tunnel key in the 24 bit VNI.

Here we are talking about network ids which corresponds to logical datapath (switch or router) tunnel key.

For now I'll leave it to 4096 and this can be revisited.

Reply:
Not sure I understand. I would expect the VNI to carry both the datapath tunnel key and the port tunnel key at the same time. Maybe it's using different VNI offsets for that. Anyway, agreed, let's leave it at 4096 for now.